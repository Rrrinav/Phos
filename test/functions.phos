fn factorial(n: i64) -> i64 {
    if (n <= 1) {
        return 1;
    }
    return n * factorial(n - 1);
}
print("Factorial of 5 is " + (factorial(5) as string));

let multiplier := 10;
let multiply_by_ten: |i64| -> i64 = |x: i64| -> i64 {
    return x * multiplier;
};

fn apply(const op: |i64| -> i64, value: i64) {
    print("Applying closure: " + (op(value) as string));
}

apply(multiply_by_ten, 7);
