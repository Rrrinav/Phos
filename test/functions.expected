Factorial of 5 is 120
Applying closure: 70
